kv/kvserver/batcheval: TestDeleteRangeTombstoneSetsGCHint failed

kv/kvserver/batcheval.TestDeleteRangeTombstoneSetsGCHint [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/18332063?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/18332063?buildTab=artifacts#/) on release-23.2 @ [7b975f24af7104c904a60f5bfb2ac68344dc2d50](https://github.com/cockroachdb/cockroach/commits/7b975f24af7104c904a60f5bfb2ac68344dc2d50):


```
=== RUN   TestDeleteRangeTombstoneSetsGCHint
    test_log_scope.go:165: test logs captured to: /artifacts/tmp/_tmp/e1ab32a24518a46d10d3b8d78552a36a/logTestDeleteRangeTombstoneSetsGCHint4265192009
    test_log_scope.go:76: use -show-logs to present logs inline
    test_server_shim.go:154: automatically injected an external process virtual cluster under test; see comment at top of test_server_shim.go for details.
    conditional_wrap.go:185: 
        pkg/kv/kvserver/batcheval_test/pkg/kv/kvserver/batcheval/cmd_delete_range_gchint_test.go:60: (TestDeleteRangeTombstoneSetsGCHint)
        	NOTICE: .LookupRange() called via implicit interface StorageLayerInterface;
        HINT: consider using .StorageLayer().LookupRange() instead.
    cmd_delete_range_gchint_test.go:72: aborted in DistSender: result is ambiguous: node unavailable; try another peer
    panic.go:523: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/e1ab32a24518a46d10d3b8d78552a36a/logTestDeleteRangeTombstoneSetsGCHint4265192009
--- FAIL: TestDeleteRangeTombstoneSetsGCHint (12.18s)
```

Parameters:
 - <code>TAGS=bazel,gss,deadlock</code>
 - <code>stress=true</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/kv
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestDeleteRangeTombstoneSetsGCHint.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-45890

kv/kvserver/batcheval.TestDeleteRangeTombstoneSetsGCHint [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/18547140?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/18547140?buildTab=artifacts#/) on release-23.2 @ [4ee96fb74079ae5e8a803151fdda784ccaace07d](https://github.com/cockroachdb/cockroach/commits/4ee96fb74079ae5e8a803151fdda784ccaace07d):


```
=== RUN   TestDeleteRangeTombstoneSetsGCHint
    test_log_scope.go:165: test logs captured to: /artifacts/tmp/_tmp/e1ab32a24518a46d10d3b8d78552a36a/logTestDeleteRangeTombstoneSetsGCHint2274231694
    test_log_scope.go:76: use -show-logs to present logs inline
    test_server_shim.go:154: automatically injected an external process virtual cluster under test; see comment at top of test_server_shim.go for details.
    conditional_wrap.go:185: 
        pkg/kv/kvserver/batcheval_test/pkg/kv/kvserver/batcheval/cmd_delete_range_gchint_test.go:60: (TestDeleteRangeTombstoneSetsGCHint)
        	NOTICE: .LookupRange() called via implicit interface StorageLayerInterface;
        HINT: consider using .StorageLayer().LookupRange() instead.
    cmd_delete_range_gchint_test.go:72: aborted in DistSender: result is ambiguous: node unavailable; try another peer
    panic.go:523: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/e1ab32a24518a46d10d3b8d78552a36a/logTestDeleteRangeTombstoneSetsGCHint2274231694
--- FAIL: TestDeleteRangeTombstoneSetsGCHint (12.32s)
```

Parameters:
 - <code>TAGS=bazel,gss,deadlock</code>
 - <code>stress=true</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestDeleteRangeTombstoneSetsGCHint.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/pkg/kv/kvserver/batcheval/cmd_delete_range_gchint_test.go
+++ b/pkg/kv/kvserver/batcheval/cmd_delete_range_gchint_test.go
@@ -26,6 +26,9 @@ func TestDeleteRangeTombstoneSetsGCHint(t *testing.T) {
 
 	ctx := context.Background()
 	srv := serverutils.StartServerOnly(t, base.TestServerArgs{
+		// Disable tenant testing here to simplify the creation of the scratch range
+		// below.
+		DefaultTestTenant: base.TestIsSpecificToStorageLayerAndNeedsASystemTenant,
 		Knobs: base.TestingKnobs{
 			Store: &kvserver.StoreTestingKnobs{
 				DisableMergeQueue: true,
@@ -36,17 +39,17 @@ func TestDeleteRangeTombstoneSetsGCHint(t *testing.T) {
 	defer srv.Stopper().Stop(ctx)
 
 	s := srv.ApplicationLayer()
+	key, err := srv.ScratchRange()
+	require.NoError(t, err, "failed to create scratch range")
 
 	store, err := srv.StorageLayer().GetStores().(*kvserver.Stores).GetStore(srv.StorageLayer().GetFirstStoreID())
 	require.NoError(t, err)
-
-	key := append(s.Codec().TenantPrefix(), roachpb.Key("b")...)
-	content := []byte("test")
-
 	repl := store.LookupReplica(roachpb.RKey(key))
+
 	gcHint := repl.GetGCHint()
 	require.True(t, gcHint.LatestRangeDeleteTimestamp.IsEmpty(), "gc hint should be empty by default")
 
+	content := []byte("test")
 	pArgs := &kvpb.PutRequest{
 		RequestHeader: kvpb.RequestHeader{
 			Key: key,

